examples/energy-management-app/linux/main.cpp

```diff
@@ -26,28 +27,36 @@ using namespace chip::app;
 using namespace chip::app::Clusters;
 using namespace chip::app::Clusters::DeviceEnergyManagement;
 using namespace chip::app::Clusters::DeviceEnergyManagement::Attributes;
+using namespace chip::app::Clusters::WaterHeaterManagement;
 
 // Parse a hex (prefixed by 0x) or decimal (no-prefix) string
 static uint32_t ParseNumber(const char * pString);
 
 // Parses the --featureMap option
-static bool FeatureMapOptionHandler(const char * aProgram, chip::ArgParser::OptionSet * aOptions, int aIdentifier,
-                                    const char * aName, const char * aValue);
+static bool EnergyAppOptionHandler(const char * aProgram, chip::ArgParser::OptionSet * aOptions, int aIdentifier,
+                                   const char * aName, const char * aValue);
 
-constexpr uint16_t kOptionFeatureMap = 'f';
+constexpr uint16_t kOptionApplication = 0xffd0;
+constexpr uint16_t kOptionFeatureMap  = 0xffd1;
+
+constexpr const char * kEvseApp = "evse";
```

Comment:
I see we are moving forward with the whole multiple apps under the same folder structure approach.
This leaves us with the choice to either 

- have every appliance app live under the energy management folder, which wouldn't make sens for matter app that don't support the device, therefore having 2 copies of each app that wants to support the energy management.
- Have different structures between EVSE + water heater vs the Apps that existed in the repo before.
- Refactor the curent folder structure to have separate folder for the common energy management files and have a separate example app for EVSE and Water heater.

I understand this is time sensitive, but we should have a followup issue somewhere to refactor this architecture, so at least we can keep in mind we should at least consider how to unify this app with the rest of the example apps in the repo.

Reply:
Agree - I think if we can have an energy management 'library' then this would look different. 

As mentioned in a previous PR - the longer term vision is to provide different energy management examples since the use cases for a Washing Machine / Dishwasher 'forecast' is very different from a Battery / EV.

We are very much at the beginning of a journey on energy management and over time this will need to continue to evolve with the ability to use the same DEM cluster with optimisers using Time of Use tariff information, and battery and EV using the power adjustment all as library calls.

I suggest sometime after SVE we workshop this since there are counter arguments against the current approach too. I'm not sure we need a follow up since the two approaches will need to be resolved in the longer term and won't be forgotten about.

Reply:
Resolving so the PR can merge